### PR TITLE
Use origin-ansible-service-broker docker image

### PR DESCRIPTION
**Pull request by djzager**
Make it so the broker template and project documentation refer to the `origin-ansible-service-broker` docker image.

### CODE REVIEW

**Comment by djzager** [comment]
# Tests
## Running `make build-image`

```
➜  ansible-service-broker git:(use-origin) ✗ make build-image
cp broker build/broker
docker build -f "/home/dzager/go/src/github.com/openshift/ansible-service-broker/build"/Dockerfile-src -t docker.io/ansibleplaybookbundle/origin-ansible-service-broker:latest "/home/dzager/go/src/github.com/openshift/ansible-service-broker/build"
Sending build context to Docker daemon  31.9 MB
Step 1/13 : FROM centos:7
 ---> 328edcd84f1b
...
Step 13/13 : ENTRYPOINT entrypoint.sh
 ---> Running in ed84c84df780
 ---> 31252431f8f4
Removing intermediate container ed84c84df780
Successfully built 31252431f8f4

Remember you need to push your image before calling make deploy
    docker push docker.io/ansibleplaybookbundle/origin-ansible-service-broker:latest
```

## Running `scripts/run_latest_build.sh` manually

```
➜  ansible-service-broker git:(use-origin) ✗ PUBLIC_IP="$(ip addr show docker0 | grep -Po 'inet \K[\d.]+')"
➜  ansible-service-broker git:(use-origin) ✗ HOSTNAME=${PUBLIC_IP}.nip.io
➜  ansible-service-broker git:(use-origin) ✗ ROUTING_SUFFIX="${HOSTNAME}"
➜  ansible-service-broker git:(use-origin) ✗ oc cluster up --image=openshift/origin --version=v3.6.0-rc.0 --service-catalog=true --routing-suffix=${ROUTING_SUFFIX} --public-hostname=${HOSTNAME}

Starting OpenShift using openshift/origin:v3.6.0-rc.0 ...
OpenShift server started.

➜  ansible-service-broker git:(use-origin) ✗ oc login -u system:admin
Logged into "https://127.0.0.1:8443" as "system:admin" using existing credentials.

➜  ansible-service-broker git:(use-origin) ✗ oc new-project ansible-service-broker
Now using project "ansible-service-broker" on server "https://127.0.0.1:8443".

➜  ansible-service-broker git:(use-origin) ✗ DOCKERHUB_USER="changeme"
➜  ansible-service-broker git:(use-origin) ✗ DOCKERHUB_PASS="changeme"
➜  ansible-service-broker git:(use-origin) ✗ DOCKERHUB_ORG="ansibleplaybookbundle"
➜  ansible-service-broker git:(use-origin) ✗ ENABLE_BASIC_AUTH="false"
➜  ansible-service-broker git:(use-origin) ✗ oc process -f templates/deploy-ansible-service-broker.template.yaml \
    -n ansible-service-broker \
    -p DOCKERHUB_USER="$DOCKERHUB_USER" \
    -p DOCKERHUB_PASS="$DOCKERHUB_PASS" \
    -p DOCKERHUB_ORG="$DOCKERHUB_ORG" \
    -p ENABLE_BASIC_AUTH="$ENABLE_BASIC_AUTH" | oc create -f -
service "asb" created
serviceaccount "asb" created
clusterrolebinding "asb" created
route "asb-1338" created
persistentvolumeclaim "etcd" created
deploymentconfig "asb" created
secret "asb-auth-secret" created
configmap "broker-config" created

➜  ansible-service-broker git:(use-origin) ✗ oc get pods -n ansible-service-broker asb-1-tgrj3 -o yaml
apiVersion: v1
kind: Pod
metadata:
...
spec:
  containers:
  - env:
    - name: BROKER_CONFIG
      value: /etc/ansible-service-broker/config.yaml
    - name: INSECURE
      value: "false"
    image: ansibleplaybookbundle/origin-ansible-service-broker:latest
    imagePullPolicy: IfNotPresent
    name: asb
    ...
```

**Comment by jmontleon** [comment]
ACK

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -136,11 +136,11 @@ See what work is in progress, upcoming, and planned via our [Trello Board](https
 
 # Download images from Docker Hub
 
-Docker images are available on [Docker Hub](https://hub.docker.com/r/ansibleplaybookbundle/ansible-service-broker/)
+Docker images are available on [Docker Hub](https://hub.docker.com/r/ansibleplaybookbundle/origin-ansible-service-broker/)
 
 * [Docker Hub published APBs](https://hub.docker.com/u/ansibleplaybookbundle/)
 
-Image [tags](https://hub.docker.com/r/ansibleplaybookbundle/ansible-service-broker/tags):
+Image [tags](https://hub.docker.com/r/ansibleplaybookbundle/origin-ansible-service-broker/tags):
```
Comment (jmrodri) [inline CONTRIBUTING.md]: +1 to contributing changes

File: Makefile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 REGISTRY         ?= docker.io
 ORG              ?= ansibleplaybookbundle
 TAG              ?= latest
-BROKER_IMAGE     ?= $(REGISTRY)/$(ORG)/ansible-service-broker
+BROKER_IMAGE     ?= $(REGISTRY)/$(ORG)/origin-ansible-service-broker
```
Comment (jmrodri) [inline Makefile]: I'm wondering if we'll need to make this a variable in case we need to make this `ose-` one day.

File: docs/deployment.md [hunk 3]
```diff
@@ -6,7 +6,7 @@ The following are the configurable parameters that you can set for deploying the
 Name | Default Value | Description
 ---|---|---
 PROJECT | ansible-service-broker | Project Namespace
-BROKER_IMAGE | ansibleplaybookbundle/ansible-service-broker:latest| Container Image to use for Ansible Service Broker in format of imagename:tag 
+BROKER_IMAGE | ansibleplaybookbundle/origin-ansible-service-broker:latest| Container Image to use for Ansible Service Broker in format of imagename:tag
```
Comment (jmrodri) [inline docs/deployment.md]: good

File: scripts/run_latest_build.sh [hunk 4]
```diff
@@ -88,7 +88,7 @@ fi
 
 ASB_ROUTE=`oc get routes | grep ansible-service-broker | awk '{print $2}'`
 
-cat <<EOF > ansible-service-broker.broker
+cat <<EOF | oc create -f -
```
Comment (jmrodri) [inline scripts/run_latest_build.sh]: ❗️ good optimization

File: templates/deploy-ansible-service-broker.template.yaml [hunk 5]
```diff
@@ -201,7 +201,7 @@ parameters:
 - description: Container Image to use for Ansible Service Broker in format of imagename:tag
   displayname: Ansible Service Broker Image
   name: BROKER_IMAGE
-  value: ansibleplaybookbundle/ansible-service-broker:latest
+  value: ansibleplaybookbundle/origin-ansible-service-broker:latest
```
Comment (jmrodri) [inline templates/deploy-ansible-service-broker.template.yaml]: hopefully, this doesn't have to be swapped out for ose- later. I think we're ok for now.

**Comment by jmrodri** [comment]
@djzager thanks for the test plan!

**Comment by djzager** [comment]
[Upstream changes for 'origin-' prefix to broker image. origin-ansible-service-broker](https://trello.com/c/msbb5oxR/373-upstream-changes-for-origin-prefix-to-broker-image-origin-ansible-service-broker)

**Pull request merged**